cmd/internal: elide issue assignee based on authorship

irfansharif opened the pull request:
Touches #51653. We've been spamming Andrei because of his refactor of everything related to roach{test/prod}, which makes it difficult to disambiguate which issues should actually be assigned to him vs. not. We replace roachtest assignment with new labels instead that hopefully are a bit more useful going forward. Unit test flakes still default to using `git blame` style ownership, but that's a yak we can shave elsewhere.

Release note: None

irfansharif committed:
cmd/internal: elide roachtest issue assignee based on git blame

Touches https://github.com/cockroachdb/cockroach/issues/51653. We've
been spamming Andrei because of his refactor of everything related to
roach{test/prod}, which makes it difficult to disambiguate which issues
should actually be assigned to him vs. not. We replace roachtest
assignment with new labels instead that hopefully are a bit more useful
going forward. Unit test flakes still default to using `git blame` style
ownership, but that's a yak we can shave elsewhere.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51660)
<!-- Reviewable:end -->

jordanlewis reviewed (changes requested):
Thumbs down on this because AFAIK it will also affect non-roachtest failures, and I don't think we've decided that those are bad yet as well.

Would you be able to only elide the issue assignment if you detect that it's `roachtest` itself failing?

irfansharif reviewed (commented):
Wanna take another look? Should be just roachtests now.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51660#-:-MCn-C7PBiffrcpa0Uo7:bwrar8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

jordanlewis committed:
roachtest: assign issues to owner projects

This commit adds project ownership assigment for roachtest with owners.
It uses a hardcoded list of project column ids to do this, which is
horrible, but I don't know of a better way.

Release note: None

irfansharif commented on pkg/cmd/roachtest/test_registry.go:
```diff
@@ -44,19 +44,50 @@ const (
-	OwnerSQLSchema:    {SlackRoom: `sql-schema`, ContactEmail: `lucy@cockroachlabs.com`},
-	OwnerStorage:      {SlackRoom: `storage`, ContactEmail: `peter@cockroachlabs.com`},
+	OwnerAppDev: {SlackRoom: `app-dev`, ContactEmail: `rafi@cockroachlabs.com`,
+		TriageColumnID: 8532151,
```
Nice jankiness. Maybe just add a comment next to each one naming the specific board/column.

irfansharif reviewed (commented):
Reviewing my own PR here but LGTM, let's get another pair of eyes on it given @tbg is out for a few days and get it in?

jordanlewis reviewed (approved):
LGTM :trollface:

irfansharif commented:
bors r+

irfansharif commented:
bors r+
